### DISCUSSION TITLE
Need help with pico w webserver controlling ws2812b's

### DISCUSSION THREAD

**Discussion by handofiron** [micropython/micropython#10006] [Using MicroPython]
Hi all, I'm trying to get a pico to control led's in my living room basically id like html web page buttons and physical buttons. Everything works but as soon as I press turn led on the loop starts and the page crashes and wont let hit led off. Any help would be appreciated my problems start at line 112 i need the if led on to loop unless i press led off.

Cheers.

https://github.com/handofiron/micropython-pico/blob/main/control-panel-new.py

**Comment by GM-Script-Writer-62850** [comment]
`if led == 8 ` if you just want to know if a string is in the url use if `led > 0`

line 117 sets led off then line 118 turns it on. why? just use `led.value(1)` and  be done with it

`led.value()` will give you the current state (0 = off; 1 = on)
`led.toggle()` is just a clean way of doing `led.value(not led.value())`

you will probably want to make use of stuff like _thread, uasyncio, and microdot, your current script will be frozen at waiting for a connection and will not be able to poll input buttons (maybe you want to use interrupts?)


your imports are a mes, you do not need all that mess and duplicate lines

you are not using threading, dht, framebuf, ssd1306, and you are not using I2C, ADC from machine

you also defined red then defined it as the same thing on the next line

**Reply by handofiron** [reply]
Line 117 just makes sure the led is off before turning on my import list is a mess as this will be merged with another project. The led turns on and off just fine. It's the ws2182bs I can't get right. As you say as soon as a loop is called the client freezes.

**Reply by jimmo** [reply]
@handofiron It doesn't look like you're using the "official" Neopixel driver. There's no `set_pixel` method on the MicroPython one.

@GM-Script-Writer-62850 is right though, this would be much simpler to use a HTTP server library. microdot is a good one to use.
